src/utils/misc.ts

```diff
@@ -35,3 +35,36 @@ export const generateRandomAlphaNumericString = (size: number) =>
   Array.from(Array(size), () =>
     Math.floor(Math.random() * 36).toString(36)
   ).join("");
+
+/**
+ * Repeatedly checks a condition function until it returns true,
+ * then resolves the returned promise. If the condition function does not return true
+ * within the specified timeout, the promise is rejected.
+ *
+ * @param conditionFunction - A function that returns a boolean value.
+ *                            The promise resolves when this function returns true.
+ * @param timeout - An optional timeout in seconds. The promise is rejected if the
+ *                  condition function does not return true within this time.
+ * @returns A promise that resolves once the conditionFunction returns true or rejects if timed out.
+ */
+export const until = (
+  conditionFunction: () => boolean,
+  timeout?: number
```

Comment:
```suggestion
  timeoutSeconds?: number
```

Is this in seconds as stated  in the jsdoc above? It looks like it gets multiplied by 1000 below

Reply:
Fixed, thanks!